__tests__/app.test.js

```diff
@@ -229,6 +229,79 @@ describe('GET Requests', () => {
 });
 
 describe('POST, PATCH & DELETE', () => {
+
+    describe('POST /api/articles/:article_id/comments', () => {
+        const newComment = {
+                username: 'rogersop',
+                body: 'Big fan, I truly love articles such as these.'
+            };
+        test('should return 201 and the posted comment', () => {
+            return request(app)
+                .post('/api/articles/1/comments')
+                .send(newComment)
+                .expect(201)
+                .then(({ body }) => {
+                    expect(body.comments[0]).toMatchObject({
+                        comment_id: expect.any(Number),
+                        body: 'Big fan, I truly love articles such as these.',
+                        article_id: 1,
+                        author: 'rogersop',
+                        votes: 0,
+                        created_at: expect.any(String)
+                    });
+                });
+        });
+        test('should return 404: "Not Found" if the article id doesnt match anything', () => {
+            return request(app)
+                .post('/api/articles/115/comments')
+                .send(newComment)
+                .expect(404)
+                .then(({ body }) => {
+                    expect(body.msg).toBe('Not Found');
+                });
+        });
+        test('should return 400: "Bad Request" if the article id doesnt match anything', () => {
+            return request(app)
+                .post('/api/articles/globaltapas/comments')
+                .send(newComment)
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toBe('Bad Request');
+                });
+        });
+        test('should return 406: "Not Acceptable" if the request is missing information', () => {
+            const halfComment = {username: 'rogersop'};
+            return request(app)
+                .post('/api/articles/1/comments')
+                .send(halfComment)
+                .expect(406)
+                .then(({ body }) => {
+                    expect(body.msg).toBe('Not Acceptable');
+                });
+        });
+        test('should return 406: "Not Acceptable" if the request has invalid entries', () => {
```

Comment:
406? Reckon could be a 400

Reply:
Also this should be accepted. The task says that unnecessary properties should just be ignored, and the comment accepted